src/components/Tooltip/index.js

```diff
@@ -144,10 +126,23 @@ class Tooltip extends PureComponent {
      * Display the tooltip in an animation.
      */
     showTooltip() {
-        Animated.timing(this.animation, {
-            toValue: 1,
-            duration: 140,
-        }).start();
+        // We have to dynamically calculate the position here as tooltip could have been rendered on some elments
+        // that has changed its position
+        this.getWrapperPosition()
+            .then(({
+                x, y, width, height,
+            }) => {
+                this.setState({
+                    wrapperWidth: width,
+                    wrapperHeight: height,
+                    xOffset: x,
+                    yOffset: y,
+                });
+                Animated.timing(this.animation, {
+                    toValue: 1,
+                    duration: 140,
+                }).start();
+            });
```

Comment:
@roryabraham I need some advice here. When I move the `Animated.timing` inside the `then` callback. The tooltip sometimes remains visible after moving out the mouse. but It does not have this behaviour when I put this adjacent after then callback.
I think due to the start animation is async now it is possible that it will start after the close animation is fired resulting in the tooltip being visible even on the mouse out. What can I do to make sure it does not fire if we moused has moved out.

Reply:
LOL. I am not able to reproduce this behavior.

Reply:
Hmmm I'm wondering now if this change is even necessary since `onLayout` should be called whenever the page layout changes, causing the wrapper view to be re-measured, right?

Reply:
So for Modal such as search page Modal, its layout is created before is visible and thus for the tooltips. So when we mouse over the title there is not layout event but due to the tooltip position was calculated when the search page was hidden towards the right, its position is not correct. Thus we have to recalculate the position. So I had put the recalculation on the `showtooltip`. As you know, that we have to put the Animation start inside the callback otherwise it won't be calculated for the first animation. Actually, now I don't see any glitch but I think due to async start, there could be one.

Reply:
Okay, in that case maybe what you could do is:

1) Add a field like `shouldStartShowAnimation` to this component
2) `onHoverIn` set it to true
3) `onHoverOut` set it to false
4) In the `showTooltip` function, only trigger the animation if that's true, like so:

```javascript
if (this.shouldStartShowAnimation) {
    Animated.timing(this.animation, {
        toValue: 1,
        duration: 140,
    }).start();
}
```